src/withResolution.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import * as React from 'react'
+import { get } from 'lodash'
+import { displayAliases as defaultDisplayAliases } from './defaults.json'
+import type { DisplayValues } from './types'
+import { getMediaQueries } from './utils'
+import { ConfigContextPropTypes } from './configProvider'
+
+type Props = {
+  show?: DisplayValues,
+  base?: number,
+}
+
+type State = { shouldShow: boolean }
+
+export default function withResolution(Component: *) {
```

Comment:
Just a heads up: the usage of web-specific APIs will prove problematic for React Native (`window`,

Reply:
do we have a way to detect screen resolution changes that would work well with RN?

Reply:
There are built in APIs to RN that detect resolution. They're not compatible with web APIs though. 

Probably fine to go this route for now. And punt on resolution detection, since we don't even have a native solution in place 😛 .

I think the use case for a `show` api is rarer for native. Maybe something more focused on changing `size` attributes based on screen size. the `show` api can accomplish that, but it'll be a bit verbose.

Reply:
Ok for now I added a step to skip `withResolution` if `window.matchMedia` is not available and log a warning. As you suggest, we can worry about a proper solution when we actually support RN.

I imagine we may want different layouts in landscape vs portrait mode or something like that but probably not much more beyond that